src/sentry/snuba/metrics/query_builder.py

```diff
@@ -288,41 +289,22 @@ def _parse_orderby(self, query_params):
 
         return MetricsOrderBy(field, direction)
 
-    def _parse_limit(self, query_params, paginator_kwargs):
-        if self.orderby:
-            return Limit(paginator_kwargs.get("limit"))
-        else:
-            per_page = query_params.get("per_page")
-            if per_page is not None:
-                # If order by is not None, it means we will have a `series` query which cannot be
-                # paginated, and passing a `per_page` url param to paginate the results is not
-                # possible
-                raise InvalidParams("'per_page' is only supported in combination with 'orderBy'")
-            return None
+    def _parse_limit(self, paginator_kwargs):
+        limit = paginator_kwargs.get("limit")
+        return Limit(limit) if limit else None
```

Comment:
`limit` cannot actually be `None` here right? The `if` here is just for the type checker? In that case I would call `paginator_kwargs["limit"]` instead, or better yet, make `limit` and `offset` individual parameters with their own types.